src/environmentd/tests/sql.rs

```diff
@@ -2430,3 +2430,71 @@ fn test_query_on_dropped_source() {
         )
     });
 }
+
+#[test]
+fn test_dont_drop_sinks_twice() {
+    let config = util::Config::default().workers(4);
+    let server = util::start_server(config).unwrap();
+
+    let (notice_tx, mut notice_rx) = futures::channel::mpsc::unbounded();
+
+    let (create_tx, create_rx) = futures::channel::oneshot::channel();
+    let (drop_tx, drop_rx) = futures::channel::oneshot::channel();
+
+    let mut client_a = server
+        .pg_config()
+        .notice_callback(move |notice| {
+            notice_tx.unbounded_send(notice).unwrap();
+        })
+        .connect(postgres::NoTls)
+        .unwrap();
+
+    let handle_a = std::thread::spawn(move || {
+        client_a.batch_execute("CREATE TABLE t1 (a INT)").unwrap();
+        client_a.batch_execute("CREATE TABLE t2 (a INT)").unwrap();
+
+        create_tx.send(()).unwrap();
```

Comment:
This can probably all be done outside of the thread so we can avoid synchronizing over the channel.

Reply:
Good call, originally it seemed like the test was hanging until the SUBSCRIBE was dropped, which is why I moved it to a second thread. Moving everything back now and this works